CDTDatastore/CDTReplicator/CDTPullReplication.m

```diff
@@ -33,7 +34,16 @@ + (instancetype)replicationWithSource:(NSURL *)source target:(CDTDatastore *)tar
 - (instancetype)initWithSource:(NSURL *)source target:(CDTDatastore *)target
 {
     if (self = [super init]) {
-        _source = source;
+        
+        NSURLComponents * sourceComponents = [NSURLComponents componentsWithURL:source resolvingAgainstBaseURL:NO];
```

Comment:
I don't think that is possible. The source and target are different types depending on the kind of replication. The obvious solution would be a class method, however that causes the problem of not having access to the iVars and we shouldn't pass the reference to self around until `init` has finished, the other option for this would be to have awkward return type of a dictionary or array, so I think that option is out.  Adding a parameter to the the super classes' init method wouldn't be the right thing to do since it would be stripping creds and not storing the value of the URL somewhere which would be misleading to anyone reading the subclass code in isolation.

Is there another option I may have missed?